pkg/operator/api/api.go

```diff
@@ -475,6 +523,14 @@ func (a *apiServer) ComponentUpdate(in *operatorv1pb.ComponentUpdateRequest, srv
 		if c.Namespace != in.GetNamespace() {
 			return
 		}
+		if c.ObjectMeta.Namespace == security.CurrentNamespace() {
+			for s := range c.Scopes {
+				scope := c.Scopes[s]
+				if utils.IsControlPlaneService(scope) {
+					return
+				}
+			}
+		}
```

Comment:
Why are we doing this here? Control plane components must support component updates eventually to have feature parity across Dapr. It also is not appropriate to have logic gating this functionality on the operator server side, rather than on the client side.

Reply:
have updated ComponentUpdate to be same as ListComponents